### PR TITLE
DEV-2783: add unique link test

### PR DESCRIPTION
**Pull request by kulgan**
Add test for detecting duplicate link definition via name or backref. Added minor update to make produced wheel more compliant with pypi

### CODE REVIEW

**Review by mpsolano — APPROVED** [review]
Looks good!

File: tests/test_gdcdictionary.py [hunk 1]
```diff
@@ -8,6 +11,97 @@
...
+
+def extract_links(source: str, links: Sequence[Mapping[str, Any]]) -> Associations:
+    """Inspect links collected from the schema of a node and generate Associations."""
+    associations = Associations()
+    for link in links:
+        if "subgroup" in link:
+            associations.update(extract_links(source, link["subgroup"]))
+            continue
+        left = Association(name=link["name"], source=source, target=link["target_type"])
+        right = Association(name=link["backref"], target=source, source=link["target_type"])
```
Comment (qqiao2024) [inline tests/test_gdcdictionary.py]: nit: 
there is no `left` or `right`. 
change to `from`, `to` or `from_source`, `to_source`.

**Review by qqiao2024 — APPROVED** [review]
